sqlglot/transforms.py

```diff
@@ -325,17 +330,30 @@ def unnest_to_explode(expression: exp.Expression) -> exp.Expression:
 
             if isinstance(unnest, exp.Unnest):
                 alias = join_expr.args.get("alias") if is_lateral else unnest.args.get("alias")
-                udtf = exp.Posexplode if unnest.args.get("offset") else exp.Explode
+                has_multi_expr = len(unnest.expressions) > 1
+                _udtf: type[Func] = exp.Posexplode if unnest.args.get("offset") else exp.Explode
 
                 expression.args["joins"].remove(join)
 
+                if unnest_using_arrays_zip and has_multi_expr:
+                    # Modify the logic to use arrays_zip if there are multiple expressions
+                    # Build arrays_zip with nested expressions correctly
+                    unnest.set(
+                        "expressions",
+                        [exp.Anonymous(this="ARRAYS_ZIP", expressions=unnest.expressions)],
+                    )
+                    _udtf = exp.Inline
```

Comment:
That's right.

So when has_multi_expr is True, we use `inline` instead of `explode`, `lateral view explode` will get a multi-row structure, and inline can expand the structure and set the column alias, so that it can be consistent with the select expr behavior.